lib/recurly/bus.js

```diff
@@ -63,23 +61,41 @@ export class Bus extends Emitter {
     this.debug(`added recipient. Total: ${this.recipients.length}`, recipient);
   }
 
+  /**
+   * Remnoves a recipient from the bus
+   *
+   * @public
+   * @param {Window|Emitter} recipient to remove
+   * @return {Boolean} whether the removal was performed
+   */
+  remove (recipient) {
+    const pos = this.recipients.indexOf(recipient);
+    return !!(~pos && this.recipients.splice(pos, 1));
+  }
+
   /**
    * Sends a generic message to all recipients
    *
    * @public
    * @param {String} event name of the message's event
    * @param {Object} body message contents
    * @param {Object} [options]
-   * @param {Array} [options.exclude] a list of
+   * @param {Array} [options.exclude] a list of recipients to exlude from receiving the message
    */
   send (event, body, options = { exclude: [] }) {
     const { groupId } = this;
     const message = new Message(event, body, groupId);
     const { exclude } = options;
 
-    this.debug(`sending message to ${this.recipients.length} recipients`, message);
-    this.recipients.forEach(r => {
-      if (~exclude.indexOf(r)) return;
+    if (!message.event) {
+      this.debug('discarding message due to lack of event name', message);
+      return;
+    }
+
+    const recipients = this.recipients.filter(r => !~exclude.indexOf(r));
```

Comment:
Just from looking at it, I'm not sure how this filter would be working. I would think you'd have to do some kind of deep equals for the filter to be able to work correctly. I could also be misunderstanding this.

Reply:
recipients and exclude are both flat `Array`s of exact Object references, so it will look something like this.

```js
this.recipients = [instance, instance2, instance3];
const exclude = [instance2];
const recipients = this.recipients.filter(r => !~exclude.indexOf(r));
=> recipients = [instance, instance3]
```

Reply:
That makes perfect sense